src/tiled/objectselectionitem.cpp

```diff
@@ -420,6 +423,10 @@ void ObjectSelectionItem::layerChanged(Layer *layer)
     }
 }
 
+void ObjectSelectionItem::objectGroupChanged(ObjectGroup * objectGroup) {
+    updateObjectLabelColors();
```

Comment:
In fact, since this function does not do anything else than call `updateObjectLabelColors`, you can connect the `objectGroupChanged` signal directly to that function.